src/webview/markdownEngine.ts

```diff
@@ -7,39 +7,70 @@ import * as os from "os";
 import * as path from "path";
 import * as vscode from "vscode";
 import { leetCodeChannel } from "../leetCodeChannel";
+import { isWindows } from "../utils/osUtils";
 
-class MarkdownEngine {
+class MarkdownEngine implements vscode.Disposable {
 
-    private readonly engine: MarkdownIt;
-    private readonly extRoot: string; // root path of vscode built-in markdown extension
+    private engine: MarkdownIt;
+    private config: MarkdownConfiguration;
+    private listener: vscode.Disposable;
 
     public constructor() {
-        this.engine = this.initEngine();
-        this.extRoot = path.join(vscode.env.appRoot, "extensions", "markdown-language-features");
+        this.reload();
+        this.listener = vscode.workspace.onDidChangeConfiguration((event: vscode.ConfigurationChangeEvent) => {
```

Comment:
Besides, all the WebViews should register a listener, updating their html when configuration has changed.

To do it, I suggest we abstract a base WebView out of three webviews, and register the listener there.

Reply:
Yes, feel free to raise another PR after this one.